File: etlAuto/etl/ETLSys.java
package etl;

import java.io.PrintStream;
import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.ArrayList;
import java.util.Calendar;

class ETLSys
{
  String sys;
  int DataKeepPeriod;
  int LogKeepPeriod;
  int RecordKeepPeriod;

  static ArrayList<ETLSys> dbGetSys(Connection con)
  {
    ArrayList list = new ArrayList();
    String sql = "SEL ETL_System, DataKeepPeriod , LogKeepPeriod, RecordKeepPeriod FROM ETL_Sys";
    try {
      Statement s = con.createStatement();
      ResultSet r = s.executeQuery(sql);
      while (r.next()) {
        ETLSys e = new ETLSys();
        e.sys = r.getString(1);
        e.DataKeepPeriod = r.getInt(2); if (r.wasNull()) e.DataKeepPeriod = 30;
        e.LogKeepPeriod = r.getInt(3); if (r.wasNull()) e.LogKeepPeriod = 30;
        e.RecordKeepPeriod = r.getInt(4); if (r.wasNull()) e.RecordKeepPeriod = 30;
        list.add(e);
      }
      r.close(); s.close(); } catch (SQLException e) {
      HouseKeeping.dberr = true;
    }return list;
  }

  public void cleaup(Connection con) {
    Calendar c = Calendar.getInstance(); long current = System.currentTimeMillis();

    c.setTimeInMillis(current - this.DataKeepPeriod * 24 * 3600 * 1000L);
    String expiredDate = String.format("%1$tY%1$tm%1$td", new Object[] { c });

    HouseKeeping.clearupPath(ETL.Auto_home + "/DATA/complete/" + this.sys, expiredDate);
    HouseKeeping.clearupPath(ETL.Auto_home + "/DATA/fail/corrupt/" + this.sys, expiredDate);
    HouseKeeping.clearupPath(ETL.Auto_home + "/DATA/fail/bypass/" + this.sys, expiredDate);
    HouseKeeping.clearupPath(ETL.Auto_home + "/DATA/fail/duplicate/" + this.sys, expiredDate);
    HouseKeeping.clearupPath(ETL.Auto_home + "/DATA/fail/error/" + this.sys, expiredDate);

    c.setTimeInMillis(current - this.LogKeepPeriod * 24 * 3600 * 1000L);
    expiredDate = String.format("%1$tY%1$tm%1$td", new Object[] { c });
    HouseKeeping.clearupPath(ETL.Auto_home + "/LOG/" + this.sys, expiredDate);

    c.setTimeInMillis(current - this.RecordKeepPeriod * 24 * 3600 * 1000L);
    expiredDate = String.format("%1$tY-%1$tm-%1$td", new Object[] { c });
    if (ETL.isPrimaryServer) {
      ETL.ShowTime(HouseKeeping.log);
      HouseKeeping.log.println("Clean repository log for system '" + this.sys + "'");
      try {
        String sql = "DELETE FROM ETL_Received_File WHERE ETL_System = '" + this.sys + 
          "' AND ReceivedTime <= '" + expiredDate + "'";
        Statement st = con.createStatement();
        st.execute(sql); st.close();
        sql = "DELETE FROM ETL_Record_Log WHERE ETL_System = '" + this.sys + 
          "' AND RecordTime <= '" + expiredDate + "'";
        st.execute(sql); st.close();

        sql = "DELETE FROM ETL_Job_Log WHERE ETL_System ='" + this.sys + 
          "' AND StartTime <= '" + expiredDate + "'";
        st.execute(sql); st.close(); } catch (SQLException e) {
        e.printStackTrace();
      }
    }
  }
}

File: etlAuto/etl/ETLSys.py
# -*- coding: utf-8 -*-
import os
import sys
import logging
import time
import datetime
import cx_Oracle as oracle
from ETL import ETL
from HouseKeeping import HouseKeeping

logging.basicConfig(level=logging.INFO)

class ETLSys:
    sys = ''
    DataKeepPeriod = 30
    LogKeepPeriod = 30
    RecordKeepPeriod = 30
    etl = ETL()
    etl.Initialize()
    houKeep = HouseKeeping()

    def dbGetSys(sys,con):
        sysList = []
        sqlText = "select ETL_System, DataKeepPeriod , LogKeepPeriod, RecordKeepPeriod from ETL_Sys"
        try:
            cursor = con.cursor()
            cursor.execute(sqlText)
            result = cursor.fetchall()
            for resTuple in result:
                e = ETLSys()
                e.sys = resTuple[0]
                e.DataKeepPeriod = 30 if resTuple[1] is None else resTuple[1]
                e.LogKeepPeriod = 30 if  resTuple[2] is None else resTuple[2]
                e.RecordKeepPeriod = 30 if  resTuple[3] is None else resTuple[3]
                sysList.append(e)
        except oracle.DatabaseError as e:
            logging.error("Database Error Message: "+str(e.message))
            self.houKeep.dberr = True    
        return sysList

    def  cleaup(self,con):
        expiredDate = (datetime.date.today()-datetime.timedelta(days=self.DataKeepPeriod)).strftime('%Y%m%d')
        self.houKeep.clearupPath(etl.Auto_home + "/DATA/complete/" + self.sys,expiredDate)
        self.houKeep.clearupPath(etl.Auto_home + "/DATA/fail/corrupt/" + self.sys,expiredDate)   
        self.houKeep.clearupPath(etl.Auto_home + "/DATA/fail/bypass/" + self.sys,expiredDate)
        self.houKeep.clearupPath(etl.Auto_home + "/DATA/fail/duplicate/" + self.sys,expiredDate)
        self.houKeep.clearupPath(etl.Auto_home + "/DATA/fail/error/" + self.sys,expiredDate)

        expiredDate = (datetime.date.today()-datetime.timedelta(days=self.LogKeepPeriod)).strftime('%Y%m%d')
        self.houKeep.clearupPath(etl.Auto_home + "/LOG/" + self.sys, expiredDate)

        expiredDate = (datetime.date.today()-datetime.timedelta(days=self.RecordKeepPeriod)).strftime('%Y%m%d')
        self.houKeep.clearupPath(etl.Auto_home + "/LOG/" + self.sys, expiredDate)

        if etl.isPrimaryServer:
            self.houKeep.log.write("Clean repository log for system '" + self.sys + "'")
            sqlText = "DELETE FROM ETL_Received_File WHERE ETL_System = '" + self.sys + "' AND ReceivedTime <= '" + expiredDate + "'"
            cursor = con.cursor()
            try:
                cursor.execute(sqlText)
                con.commit()
            except oracle.DatabaseError as e:
                logging.error("Oracle-Error-Message: "+str(e.message))
                cursor.close()
                con.close()

            sqlText = "DELETE FROM ETL_Record_Log WHERE ETL_System = '" + self.sys + "' AND RecordTime <= '" + expiredDate + "'"
            cursor = con.cursor()
            try:
                cursor.execute(sqlText)
                con.commit()
            except oracle.DatabaseError as e:
                logging.error("Oracle-Error-Message: "+str(e.message))
                cursor.close()
                con.close()

            sqlText = "DELETE FROM ETL_Job_Log WHERE ETL_System = '" + self.sys + "' AND StartTime <= '" + expiredDate + "'"
            cursor = con.cursor()
            try:
                cursor.execute(sqlText)
                con.commit()
            except oracle.DatabaseError as e:
                logging.error("Oracle-Error-Message: "+str(e.message))
                cursor.close()
                con.close()
                
if __name__ == '__main__':
    #etl = ETL.ETL()
    #con = etl.Connect()
    #etlsys = ETLSys()
    #testList = etlsys.dbGetSys(con)
    #for test in testList:
    #    print test.sys
    #pass
    #print ETLSys.DataKeepPeriod    
    #print ETL.PrintVersionInfo('ETL AUTOMINATION')
    print ETLSys.etl.Auto_home
